pkg/signatureaggregator/signature-aggregator.go

```diff
@@ -15,11 +35,603 @@ func NewSignatureAggregatorLogger(
+		}
+		config.TrackedSubnetIDs = mergedSubnetIDs
+
+		// Merge manually tracked peers
+		existingPeers := make(map[string]PeerConfig)
+		for _, peer := range existingConfig.ManuallyTrackedPeers {
+			existingPeers[peer.ID] = peer
+		}
+		for _, peer := range config.ManuallyTrackedPeers {
+			existingPeers[peer.ID] = peer
+		}
+		mergedPeers := make([]PeerConfig, 0, len(existingPeers))
+		for _, peer := range existingPeers {
+			mergedPeers = append(mergedPeers, peer)
+		}
+		config.ManuallyTrackedPeers = mergedPeers
+
+		// Keep other existing values
+		config.LogLevel = existingConfig.LogLevel
+		config.SignatureCacheSize = existingConfig.SignatureCacheSize
+		config.AllowPrivateIPs = existingConfig.AllowPrivateIPs
+	}
+
+	// Marshal and write the config
+	configBytes, err := json.MarshalIndent(config, "", "    ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal config: %w", err)
+	}
+
+	// Create parent directory if it doesn't exist
+	if err := os.MkdirAll(filepath.Dir(configPath), 0o755); err != nil {
+		return fmt.Errorf("failed to create config directory: %w", err)
+	}
+
+	if err := os.WriteFile(configPath, configBytes, 0o600); err != nil {
+		return fmt.Errorf("failed to write config file: %w", err)
+	}
+
+	return nil
+}
+
+// SignatureAggregatorConfig represents the configuration for the signature aggregator.
+type SignatureAggregatorConfig struct {
+	LogLevel             string       `json:"log-level"`
+	PChainAPI            APIConfig    `json:"p-chain-api"`
+	InfoAPI              APIConfig    `json:"info-api"`
+	SignatureCacheSize   int          `json:"signature-cache-size"`
+	AllowPrivateIPs      bool         `json:"allow-private-ips"`
+	TrackedSubnetIDs     []string     `json:"tracked-subnet-ids"`
+	ManuallyTrackedPeers []PeerConfig `json:"manually-tracked-peers"`
+	APIPort              int          `json:"api-port"`
+	MetricsPort          int          `json:"metrics-port"`
+}
+
+// APIConfig represents the configuration for an API endpoint.
+type APIConfig struct {
+	BaseURL string `json:"base-url"`
+}
+
+// PeerConfig represents the configuration for a peer.
+type PeerConfig struct {
+	ID string `json:"id"`
+	IP string `json:"ip"`
+}
+
+func isPortAvailable(port int) bool {
+	addr := fmt.Sprintf("localhost:%d", port)
+	conn, err := net.DialTimeout("tcp", addr, 100*time.Millisecond)
+	if err != nil {
+		// If we can't connect, the port is available
+		return true
+	}
+	// If we can connect, the port is in use
+	if err = conn.Close(); err != nil {
+		// Log the error but still return false since port is in use
+		ux.Logger.RedXToUser("failed to close connection while checking port %d: %s", port, err)
+	}
+	return false
+}
+
+func generateAPIMetricsPorts() (int, int, error) {
+	// Create a context with a 30 second timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Start with default ports and increment by 2 each time
+	apiPort := 8080
```

Comment:
can you use a less standard port as a starting base? eg 9090

Reply:
8080 is the standard for signature aggregator according to their repo, why would we use 9090?

Reply:
8080 is standard port for many http services. signature aggregator just used the same standard port but
for a tool like our that deploys a signature aggregator in the background, it may be better to just use another
port to avoid unneeded conflicts (I mean, if a user install signature aggregator he may easily find the issue if there is a port conflict - he is executing that binary, and can check the doc). but if the user executes CLI, he will have issues difficult to trace.